sbc_translation/models/correspondence.py

```diff
@@ -22,22 +33,119 @@ class Correspondence(models.Model):
+        translation_id = tc.upsert_translation(text_id, letter)
+        tc.upsert_translation_status(translation_id)
+
+        # Transfert file on the NAS
+
+        # Retrieve configuration
+        smb_user = config.get('smb_user')
+        smb_pass = config.get('smb_pwd')
+        smb_ip = config.get('smb_ip')
+        smb_port = int(config.get('smb_port', 0))
+        if not (smb_user and smb_pass and smb_ip and smb_port):
+            raise Exception('No config SMB in file .conf')
+
+        # Copy file in the imported letter folder
+        smb_conn = SMBConnection(smb_user, smb_pass, 'openerp', 'nas')
+        if smb_conn.connect(smb_ip, smb_port):
+            file_ = BytesIO(base64.b64decode(
+                letter.letter_image.with_context(
+                    bin_size=False).datas))
+            config_obj = self.env['ir.config_parameter']
+            nas_share_name = (config_obj.search(
+                [('key', '=', 'sbc_translation.nas_share_name')])[0]).value
+
+            child = sponsorship.child_id
+            sponsor = sponsorship.partner_id
+            letter_name = "_".join(
+                (child.code, sponsor.ref, str(letter.id)))
+            letter_name = ".".join((letter_name, 'pdf'))
+
+            nas_letters_store_path = (config_obj.search(
+                [('key', '=', 'sbc_translation.nas_letters_store_path')])
+                [0]).value + letter_name
+            smb_conn.storeFile(nas_share_name,
+                               nas_letters_store_path, file_)
+
+            logger.info('File {} store on NAS with success'
+                        .format(letter.letter_image.name))
+
+            letter.state = 'Global Partner translation queue'
+        else:
+            raise Exception('Connection to NAS failed')
+
+    # CRON Methods
+    ##############
+    @api.model
+    def check_local_translation_done(self):
+        tc = translate_connector.TranslateConnect()
+        letters_to_update = tc.get_translated_letters()
+
+        if not letters_to_update == -1:
+            for letter in letters_to_update:
+                correspondence = self.browse(letter["letter_odoo_id"])
+                reload(sys)
+                sys.setdefaultencoding('UTF8')
+
+                tg_lang = letter["target_lang"]
+                target_lang_id = self.env['res.lang.compassion'].search(
+                    [('code_iso', '=', tg_lang)]).id
+                # find the good text to writte
+                if tg_lang == 'eng':
+                    target_text = 'english_text'
+                elif tg_lang in self.get_child_langs_code(correspondence):
+                    target_text = 'translated_text'
+                else:
+                    logger.error("Correspondence letters language no match")
+
+                # UPDATE Odoo Database
+                correspondence.write(
+                    {target_text: letter["text"],
+                     'state': 'Received in the system',
```

Comment:
Will the support of child letters be added in a next Pull request?
Because this won't work for child letters, they should be in `Published to Global Partner` state.